gecko/Dockerfile

```diff
@@ -35,8 +35,8 @@ ENV PATH $PATH:$HOME/.git-cinnabar
 # Inject the path update to the top as bash completion needs to call rustc
 RUN sed -i '1s|^|PATH=\"\$PATH:$HOME/.cargo/bin/\"\n|' $HOME/.bashrc
 
-# Delete go directories
-RUN rm -rf go-packages go
+# Delete useless directories for Firefox
+RUN rm -rf go-packages go .gem .subversion .rvm
```

Comment:
Sadly, removing stuff here will not have the desired effect.

Docker captures the entire file system state after each `RUN` instruction, and saves it into a dedicated "layer". Modifying or deleting files in a different `RUN` instruction later will just create a new layer that "masks" (or "hides") the previous layers, but the large files will remain present in Docker's layer history (just like if you add a giant file with `git` or `hg`, and then later delete the file in a new commit, the large file will remain in the repo history and continue to cause long cloning times).

In order to fully remove files from Docker images (and make the images actually lighter), we need to find in which layer or `RUN` instruction the large files are being created, and change that `RUN` instruction.

In other words:

```Dockerfile
RUN wget 30GB.tar.gz
RUN rm 30GB.tar.gz
```

will still produce a 30GB+ image.

Reply:
Oh, yeah, you see how much I am an expert on this ;)

What is your opinion about multistage builds ?
https://docs.docker.com/develop/develop-images/multistage-build/